Fix bug in two tasks incorrectly labeled as true-valid-memsafety.

mdangl opened the pull request:
The bug is already preserved in two other tasks. This fixes #521.

mutilin commented on c/ldv-memsafety/memleaks_test23_1_true-valid-memsafety.i:
```diff
@@ -1062,8 +1062,8 @@ int ldv_nonpositive(void) {
- if(r>0) return r;
- else return 1;
+ if(r>255) return r;
+ else return 256;
```
Could we put this condition to the caller place? Because the name of the function suggests that it returns >=1

mutilin commented:
We should fix a bug in the original files `ldv-memsafety/memleaks-notpreprocessed/`

mdangl commented:
Thanks @mutilin for your review. I addressed both your comments:
1. Moved restriction to values larger than 255 to caller location
2. Put the fix also in memleaks-notpreprocessed/ (I had looked for .c files alongside the .i files, but missed the subdirectory)

mdangl commented:
@mutilin assuming that the CI checks pass, would you approve this pull request now?

mutilin commented:
The code looks OK, but it seems that the benchmark now do not reflect the original intention defined in comments

mdangl commented:
Pull request not required, the tasks are actually fine.
